### PR TITLE
NEXT-9202 - Add keyboard shortcuts for reviews

### LINKED ISSUE #32: NEXT-9202 - Shortcut key does not work - but only in a specific menu
EN:
Description: The shortcut Alt+s to save does not work - but only if I want to comment or unlock ratings.
When I go to another menu, it is accepted again.

Environment: Chromium

Steps to reproduction: repeated

Expected result: works

Actual result: does not work

DE:
Beschreibung: Das Tastenkürzel Alt+s zum speichern klappt nicht - aber nur, wenn ich Bewertungen kommentieren oder freischalten möchte.
Wenn ich in ein anderes Menü gehe, wird es wieder akzeptiert.

Umgebung: Chromium

Schritte zur Reproduktion: wiederholt

Erwartetes Ergebnis: klappt

Tatsächliches Ergebnis: klappt nicht

### PR DESCRIPTION
**Pull request by aragon999**
### 1. Why is this change necessary?
Currently it is not possible to use the keyboard shortcuts for the reviews, to save and cancel the currently selected review.

### 2. What does this change do, exactly?
Add the keyboard shortcuts.

### 3. Describe each step to reproduce the issue or behaviour.
Open a review in the administration and press `ALT + S` (on my machine).

### 4. Please link to the relevant issues (if any).
Fixes #32

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [x] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: changelog/_unreleased/2020-09-17-add-keyboard-shortcuts-for-reviews.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Add keyboard shortcuts for reviews
+issue: NEXT-9202
+author: max
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-add-keyboard-shortcuts-for-reviews.md:4]: Can you please use the full name here? :-)

File: changelog/_unreleased/2020-09-17-add-keyboard-shortcuts-for-reviews.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+author_email: max@swk-web.com
+author_github: aragon999
```
Comment (PaddyS) [inline changelog/_unreleased/2020-09-17-add-keyboard-shortcuts-for-reviews.md:6]:
```suggestion
author_github: @aragon999
```

File: changelog/_unreleased/2020-09-17-add-keyboard-shortcuts-for-reviews.md (hunk 1 above)
Comment (aragon999) [reply changelog/_unreleased/2020-09-17-add-keyboard-shortcuts-for-reviews.md]: Done

File: src/Administration/Resources/app/administration/src/module/sw-review/page/sw-review-detail/index.js [hunk 3]
```diff
@@ -116,6 +150,10 @@ Component.register('sw-review-detail', {
 
         onSaveFinish() {
             this.isSaveSuccessful = false;
+        },
+
+        onCancel() {
```
Comment (taltholtmann) [inline src/Administration/Resources/app/administration/src/module/sw-review/page/sw-review-detail/index.js:155]: why is this necessary? Please revert this change if it is only cosmetic.

**Review by taltholtmann — CHANGES_REQUESTED** [review]

File: src/Administration/Resources/app/administration/src/module/sw-review/page/sw-review-detail/index.js (hunk 3 above)
Comment (aragon999) [reply src/Administration/Resources/app/administration/src/module/sw-review/page/sw-review-detail/index.js:155]: Otherwise the ESCAPE key cannot properly be mapped, see:
https://github.com/shopwareBoostDay/platform/pull/156/files#diff-b5bd3d7f08c3d2c6a4a00b2e715a1b78R25
Or not sure what exactly is meant?

Comment (taltholtmann) [reply src/Administration/Resources/app/administration/src/module/sw-review/page/sw-review-detail/index.js:155]: Hey, sry for the late answer. 
You have removed the `:routerLink="{ name: 'sw.review.index' }` of the button and added the onCancel method with router push. Wasnt it the same behaviour with the router link?

Comment (aragon999) [reply src/Administration/Resources/app/administration/src/module/sw-review/page/sw-review-detail/index.js:155]: Unfortunatly I am now on holidays, so I will need to see how good I can modifications to this PR.

But yes that is correct, I replaced the link by the function which is needed because of the shortcut mapping. In my opinion this is not only a cosmetic change, but also unifies the behavior of the back button and the shortcut, as it is the case for example in the product detail view. I can change the link again, but the `onCancel` method would still be required.

Comment (taltholtmann) [reply src/Administration/Resources/app/administration/src/module/sw-review/page/sw-review-detail/index.js:155]: Tank you for the explanation. I'm okay with this change.

**Review by taltholtmann — APPROVED** [review]
